os/mm/mm_heap/mm_check_heap_corruption.c

```diff
@@ -84,14 +86,32 @@ typedef enum node_type_e node_type_t;
 
 static void dump_node(struct mm_allocnode_s *node, node_type_t type)
 {
+#if CONFIG_TASK_NAME_SIZE > 0
+#if defined(CONFIG_BUILD_PROTECTED) && !defined(__KERNEL__)
+	char myname[CONFIG_TASK_NAME_SIZE];
+#endif
+#endif
+
 	if (type == TYPE_CORRUPTED) {
 		dbg("CORRUPTED NODE: addr = 0x%08x size = %u preceding size = %u\n", node, node->size, MM_PREV_NODE_SIZE(node));
 	} else if (type == TYPE_OVERFLOWED) {
 		dbg("OVERFLOWED NODE: addr = 0x%08x size = %u type = %c\n", node, node->size, IS_ALLOCATED_NODE(node) ? 'A' : 'F');
 	}
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
+#if CONFIG_TASK_NAME_SIZE > 0
+#if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
+	dbg("Node owner pid = %u (%s), allocated by code at addr = 0x%08x\n", node->pid, (sched_gettcb(node->pid))->name, node->alloc_call_addr);
```

Comment:
We can use `prctl` from flat and kernel also.
How about merging with belows?
I know `sched_gettcb` is much better and faster, but I think it is okay, because this is for debugging.

Reply:
Done